tests/e2e/specs/scaffold.spec.ts

```diff
@@ -0,0 +1,102 @@
+import { expect } from "@playwright/test";
+import { stubMultipleDialogs } from "electron-playwright-helpers";
+import { mkdtempSync } from "fs";
+import { tmpdir } from "os";
+import path from "path";
+import { test } from "../baseTest";
+import { TextDocument } from "../objects/editor/TextDocument";
+import { Notification } from "../objects/notifications/Notification";
+import { NotificationArea } from "../objects/notifications/NotificationArea";
+import { Quickpick } from "../objects/quickInputs/Quickpick";
+import { SupportView } from "../objects/views/SupportView";
+import { View } from "../objects/views/View";
+import { ProjectScaffoldWebview } from "../objects/webviews/ProjectScaffoldWebview";
+import { openConfluentExtension } from "./utils/confluent";
+
+/**
+ * E2E test suite for testing the Project Scaffolding functionality.
+ * {@see https://github.com/confluentinc/vscode/issues/1840}
+ */
+
+test.describe("Project Scaffolding", () => {
+  test.beforeEach(async ({ page }) => {
+    await openConfluentExtension(page);
+  });
+
+  // Templates covered by the E2E tests
+  // Mapping of display names to template names
+  const templates: Array<[string, string]> = [
+    ["Kafka Client in Go", "go-client"],
+    ["Kafka Client in Java", "java-client"],
+    ["Kafka Client in JavaScript", "javascript-client"],
+    ["Kafka Client in .NET", "dotnet-client"],
+  ];
+  
+  test.describe("Support view", () => {
+    for (const [templateName, projectDirName] of templates) {
+      test(`should generate ${templateName} template from Support view`, async ({
+        page,
+        electronApp,
+      }) => {
+        // Stub the showOpen dialog
+        // Will lead to the generated project being stored in a temp directory
+        const tmpProjectDir = mkdtempSync(path.join(tmpdir(), "vscode-test-scaffold-"));
+        await stubMultipleDialogs(electronApp, [
+          {
+            method: "showOpenDialog",
+            value: {
+              filePaths: [tmpProjectDir]
+            },
+          },
+        ]);
+        
+        // Given we navigate to the Support view
+        const supportView = new SupportView(page);
+        // and we start the generate project flow
+        const projectTreeItem = supportView.treeItems.filter({
+          hasText: "Generate Project from Template",
+        });
+        await expect(projectTreeItem).toHaveCount(1);
+        await projectTreeItem.click();
```

Comment:
@shouples, WDYT of moving this to the `View` class to make it reusable? We could add something like:

```javascript
async clickTreeItem(label: string): Promise<void> {
  const treeItem = this.treeItems.filter({ hasText: label });
  await treeItem.click();
}
```

Reply:
I think that and your `selectItem()` suggestion for `Quickpick`s make sense! Lots of room to iterate and improve here 👍